searchlib/src/vespa/searchlib/common/allocatedbitvector.cpp

```diff
@@ -29,6 +29,12 @@ std::pair<BitVector::Index, BitVector::Index>
 extract_size_capacity(const AllocatedBitVector & bv) {
     BitVector::Index size = bv.size();
     BitVector::Index capacity = bv.capacity();
+    while (capacity < size) {
+        // Since size and capacity might be changed in another thread we need
+        // this fallback to avoid inconsistency during shrink.
+        size = bv.size();
+        capacity = bv.capacity();
+    }
```

Comment:
both size() and capacity() are very small inlined methods.  Thus the compiler can just cache the result in a register. You need to add a std::atomic_thread_fence og replace the loop with `size = std::min(size, capacity);`

Reply:
Added sequential consistency.